NEMO/apps/kiosk/views.py

```diff
@@ -433,3 +449,159 @@ def get_badge_reader(request) -> BadgeReader:
+
+    try:
+        date = parse_date(request.POST["estimated_resolution_dt"])
+        estimated_resolution_time = localize(
+            datetime.combine(date, parse_time(request.POST["estimated_resolution_tm"]))
+        )
+    except:
+        estimated_resolution_time = None
+
+    if not form.is_valid():
+        errors = nice_errors(form)
+
+        dictionary["message"] = errors.as_ul()
+        dictionary["estimated_resolution_dt"] = request.POST["estimated_resolution_dt"]
+        dictionary["estimated_resolution_tm"] = request.POST["estimated_resolution_tm"]
+        if "problem_category" in form.cleaned_data:
+            dictionary["problem_category"] = form.cleaned_data["problem_category"].id
+        if "status" in form.cleaned_data:
+            dictionary["status"] = form.cleaned_data["status"].id
+        if "description" in form.cleaned_data:
+            dictionary["description"] = form.cleaned_data["description"]
+        if "safety_hazard" in form.cleaned_data:
+            dictionary["safety_hazard"] = form.cleaned_data["safety_hazard"]
+        if "force_shutdown" in form.cleaned_data:
+            dictionary["force_shutdown"] = form.cleaned_data["force_shutdown"]
+        return render(request, "kiosk/tool_report_problem.html", dictionary)
+
+    task = form.save()
+    task.estimated_resolution_time = estimated_resolution_time
+    task.save()
+
+    if not settings.ALLOW_CONDITIONAL_URLS and task.force_shutdown:
+        site_title = ApplicationCustomization.get("site_title")
+
+        dictionary["message"] = (
+            f"Tool control is only available on campus. When creating a task, you can't force a tool shutdown while "
+            f"using {site_title} off campus."
+        )
+        return render(request, "kiosk/tool_report_problem.html", dictionary)
+
+    if task.force_shutdown:
+        # Shut down the tool.
+        task.tool.operational = False
+        task.tool.save()
+        # End any usage events in progress for the tool or the tool's children.
+        UsageEvent.objects.filter(tool_id__in=task.tool.get_family_tool_ids(), end=None).update(end=timezone.now())
+        # Lock the interlock for this tool.
+        try:
+            tool_interlock = Interlock.objects.get(tool__id=task.tool.id)
+            tool_interlock.lock()
+        except Interlock.DoesNotExist:
+            pass
+
+    if task.safety_hazard:
+        concern = (
+            "This safety issue was automatically created because a "
+            + str(task.tool).lower()
+            + " problem was identified as a safety hazard.\n\n"
+        )
+        concern += task.problem_description
+        issue = SafetyIssue.objects.create(reporter=request.user, location=task.tool.location, concern=concern)
+        send_safety_email_notification(request, issue)
+
+    send_new_task_emails(request, task, None)
+    set_task_status(request, task, request.POST.get("status"), request.user)
```

Comment:
I would see if it's not too much trouble to create a function in tasks.py and reuse it here.
Maybe the function could return a message if there is an error, and nothing otherwise (to handle ALLOW_CONDITIONAL_URLS etc.)